frame/balances/src/lib.rs

```diff
@@ -1222,6 +1240,193 @@ impl<T: Config<I>, I: Instance> ReservableCurrency<T::AccountId> for Module<T, I
 	}
 }
 
+impl<T: Config<I>, I: Instance> NamedReservableCurrency<T::AccountId> for Module<T, I>  where
+	T::Balance: MaybeSerializeDeserialize + Debug
+{
+	fn reserved_balance_named(id: &ReserveIdentifier, who: &T::AccountId) -> Self::Balance {
+		let reserves = Self::reserves(who);
+		reserves
+			.binary_search_by_key(id, |data| data.id)
+			.map(|idx| reserves[idx].amount)
+			.unwrap_or_default()
+	}
+
+	/// Move `value` from the free balance from `who` to their reserved balance.
+	///
+	/// Is a no-op if value to be reserved is zero.
+	fn reserve_named(id: &ReserveIdentifier, who: &T::AccountId, value: Self::Balance) -> DispatchResult {
+		<Self as ReservableCurrency<_>>::reserve(who, value)?;
```

Comment:
hmm? you are doing a normal reserved and a named here?

Reply:
named reserve is also a normal reserve so that the value from `reserved_balance` still make sense (reserved balance + free balance == total_balance)

Reply:
ahh okay I thought they were entirely distinct, and you could only do either `Currency::reserved_balance` (which would return only unnamed reserved stuff) and `Currency::named_reserve_balance`. 

But I guess having them linked is beneficial as you can query the grand total reserved of an account which is useful.